pkg/broker/broker.go

```diff
@@ -264,8 +264,8 @@ func addNameAndIDForSpec(specs []*apb.Spec, registryName string) {
 		// need to make / a hyphen to allow for global uniqueness
 		// but still match spec.
 
-		imageName := strings.Replace(spec.Image, ":", "-", -1)
-		spec.FQName = strings.Replace(fmt.Sprintf("%v-%v", registryName, imageName),
+		spec.FQName = strings.Replace(
+			fmt.Sprintf("%v-%v", registryName, spec.FQName),
```

Comment:
I worry that the FQName could have `.` in it. If it does, this has caused issues with the service catalog in the past and might be worth replacing here.